src/Pirouette/Term/Syntax.hs

```diff
@@ -41,7 +41,7 @@ separateBoundFrom u t =
 
     -- `structuredInter` transforms the set of name clashes into a map
     -- from nameString to the list of nameUnique associated.
-    structuredInter inter = Set.fold (\n m -> Map.insertWith (++) (nameString n) [nameUnique n] m) Map.empty inter
+    structuredInter = Set.fold (\n m -> Map.insertWith (++) (nameString n) [nameUnique n] m) Map.empty
```

Comment:
I am not necessarily a big fan of eta-reduction. I don't think it brings more clarity.